addons/web/static/src/legacy/js/libs/bootstrap.js

```diff
@@ -148,3 +148,27 @@ ScrollSpy.prototype._activate = function (target) {
     }
     bootstrapSpyActivateFunction.apply(this, arguments);
 };
+
+/* Bootstrap modal scrollbar compensation on non-body */
+const bsAdjustDialogFunction = Modal.prototype._adjustDialog;
+Modal.prototype._adjustDialog = function () {
+    const document = this._element.ownerDocument;
+    document.body.classList.remove('modal-open');
+    const $scrollable = $().getScrollingElement(document);
+    if (document.body.contains($scrollable[0])) {
+        $scrollable.compensateScrollbar(true);
+    }
+    document.body.classList.add('modal-open');
+    return bsAdjustDialogFunction.apply(this, arguments);
+};
+
+const bsResetAdjustmentsFunction = Modal.prototype._resetAdjustments;
+Modal.prototype._resetAdjustments = function () {
+    const document = this._element.ownerDocument;
+    document.body.classList.remove('modal-open');
```

Comment:
Unlike in the function above, this removed class is not restored after the call to `compensateScrollbar`. Is it intentional ?

Reply:
> Unlike in the function above, this removed class is not restored after the call to `compensateScrollbar`. Is it intentional ?

Yes, it is. The `_resetAdjustments` function is called when the modal is closed. Therefore, the `modal-open` class must be removed from the body. Bootstrap removes the class in `_hideModal` (in the modal.js file) just before calling `_resetAdjustments`. However, since `_resetAdjustments` is now also called in the new `noBackdropPopupWidget` widget, it is necessary to ensure that the class is also removed for this case.